src/Button/Button.component.tsx

```diff
@@ -3,113 +3,619 @@ import * as React from 'react';
+        icon &&
+        css`
+            ${iconOnly
+                ? css`
+                      // Make it a square
+                      width: ${dimensions[size].height}rem;
+                      min-width: initial;
+                      padding: 0;
+                  `
+                : css`
+                      // Space icon from text
+                      & > .anchor-icon {
+                          margin-right: ${size === 'xlarge' || size === 'large'
+                              ? 0.5
+                              : 0.375}rem;
+                      }
+                  `}
+        `}
+`;
+
+interface StyledHitboxProps {
+    buttonHeight: number;
+    buttonWidth?: number;
+}
+
+const StyledHitbox = styled.div<StyledHitboxProps>`
+    position: absolute;
+
+    // overlap border
+    top: -1px;
+    left: -1px;
+    right: -1px;
+    bottom: -1px;
+
+    // expand using margin to get to 3 rem tall and wide
+    margin: ${({ buttonHeight, buttonWidth }: StyledHitboxProps) =>
+        `-${(3 - buttonHeight) / 2}rem ${
+            buttonWidth && buttonWidth < 3
+                ? `-${(3 - buttonWidth) / 2}rem`
+                : '0'
+        }`};
 `;
 
 export const Button = ({
     className,
+    flip = false,
+    variant = 'primary',
+    size = 'large',
+    colorTheme,
+    reverse,
+    circular,
+    children,
+    icon: Icon,
     ...props
-}: ButtonProps): React.ReactElement<any> => (
-    <StyledButton
-        className={classNames('anchor-button', className)}
-        {...props}
-    />
-);
+}: ButtonProps): React.ReactElement<any> => {
+    const iconOnly = Icon && React.Children.count(children) === 0;
+
+    const iconScale = iconOnly
+        ? size === 'xsmall' ||
+          (size === 'small' && (circular || variant === 'minimal'))
+            ? 'md'
+            : 'lg'
+        : 'md';
+
+    const height = dimensions[size].height;
+    const borderRadius = circular
+        ? `${height / 2}rem`
+        : sizes.border.radius.base;
+    const width = iconOnly ? dimensions[size].height : dimensions[size].width;
+
+    // find theme from string, otherwise use defaults
+    if (typeof colorTheme === 'string') {
+        colorTheme = colors[colorTheme];
+    }
+    if (!colorTheme) {
+        colorTheme = reverse
+            ? reverseDefaults[variant]
+            : themeDefaults[variant];
+    }
+
+    const buttonStyles = ButtonColorStyles({ colorTheme, reverse })[variant];
+
+    return (
+        <StyledButton
+            className={classNames('anchor-button', className)}
+            flip={flip}
+            size={size}
+            colorTheme={colorTheme}
+            reverse={reverse}
+            height={height}
+            icon={Icon}
+            iconOnly={iconOnly}
+            circular={circular}
+            variant={variant}
+            borderRadius={borderRadius}
+            buttonStyles={buttonStyles}
+            {...props}
+        >
+            {(height < 3 || width < 3) && (
```

Comment:
If the width was set to below 3rem, but the button ended up being wider (because wide contents) then a horizontal hitbox could be added to the button when it isn't needed.

In practice, the only kosher button that is below 3 rem is the icon-only button, which isn't at risk of being expanded past its defined width.

Reply:
Worth adding a note in a usage guideline in the docs.